tasks/main.yml

```diff
@@ -54,6 +54,21 @@
   notify:
     - restart keepalived
 
+- name: Configure keepalived extra params
+  lineinfile:
+    line: "{{ item }}"
+    regexp: "^{{ item.split('=')[0] }}"
+    dest: "{{ keepalived_daemon_options_file_path }}"
+    state: present
+  with_items: "{{ keepalived_daemon_default_options_overrides }}" 
+  when:
+    - keepalived_daemon_default_options_overrides is defined
+    - keepalived_daemon_options_file_path is defined
```

Comment:
if you move keepalived_daemon_options_file_path to vars/, as said above, both will always be true.

Because the list "keepalived_daemon_default_options_overrides" is empty by default [], this task would do nothing, which is fine, so you could remove the conditionals.